### PR TITLE
Remove multi-tenant path check in auth handler

### PR DESCRIPTION
**Pull request by zengyan-amazon**
### Description

The Opensearch Dashboards tenant only impacts the saved objects access. The multi-tenant path check in the auth handler was meant to by pass the resolve tenant logic for some APIs which doesn't deal with saved objects. However with the new datasource support feature in OpenSearch Dashboards, Dashboards APIs which used to not dealing with saved objects such as data queries and index resolve operations will require the new data source saved object type under the hood.

This change removes the multi-tenant path check, so that tenant header will be injected to all route handlers. So that all route handlers can work with the right tenant in case they need to deal with any saved objects.

Signed-off-by: Yan Zeng <zengyan@amazon.com>

### Category
Bug fix

### Why these changes are required?
This change is required for OpenSearch Dashboards data source support feature to work with correct tenant when querying remote data sources.

### What is the old behavior before changes and new behavior after changes?
Old behavior: auth handler will not inject tenant header for some APIs (e.g. APIs starts with `/internal`)
New behavior: auth handler will inject tenant header to all OSD APIs

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2634

### Testing
Manually tested
Old behavior causing error getting saved objects:
image

New behavior solved the issue:
image

### Check List
- [x] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by kristenTian — DISMISSED** [review]

**Review by cliu123 — APPROVED** [review]

**Review by DarshitChanpura — DISMISSED** [review]
Ty for submitting this PR. Could you please add a test/modify a test to check this behavior?

File: server/auth/types/authentication_type.test.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  it(`internal API includes tenant info`, async () => {
+    const request = httpServerMock.createOpenSearchDashboardsRequest({
+      path: '/internal/v1',
+    });
+    const response = jest.fn();
+    const toolkit = {
+      authenticated: jest.fn((value) => value),
+    };
+    const result = await dummyAuthType.authHandler(request, response, toolkit);
+    console.log(result);
```
Comment (cliu123) [inline server/auth/types/authentication_type.test.ts:85]: Please remove this.

Comment (zengyan-amazon) [reply server/auth/types/authentication_type.test.ts]: removed debug output :)

**Review by cliu123 — DISMISSED** [review]

**Review by kristenTian — APPROVED** [review]

**Review by cwperks — DISMISSED** [review]

**Comment by cliu123** [comment]
Integration test failed. `this.esClient.asScoped is not a function`, but [the function](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/server/opensearch/legacy/cluster_client.ts#L208) is there.
```
FAIL server/auth/types/authentication_type.test.ts
  ● test tenant header › common API includes tenant info

    this.esClient.asScoped is not a function

      113 |         });
      114 |     } catch (error) {
    > 115 |       throw new Error(error.message);
          |             ^
      116 |     }
      117 |   }
      118 |

      at SecurityClient.authinfo (plugins/security-dashboards-plugin/server/backend/opensearch_security_client.ts:115:13)
      at DummyAuthType.authHandler (plugins/security-dashboards-plugin/server/auth/types/authentication_type.ts:207:44)
      at Object.<anonymous> (plugins/security-dashboards-plugin/server/auth/types/authentication_type.test.ts:86:20)

**Review by cliu123 — APPROVED** [review]

**Review by cwperks — APPROVED** [review]

**Pull request merged**